COMMANDS.org

```diff
@@ -0,0 +1,117 @@
+#+TITLE: Nix Command Reference
+#+AUTHOR: github.com/Pitta
+
+A reference to ~nix~ commands found to be helpful in day to day use.
+
+* Package Management
+
+Coming from package managers like ~brew~, ~port~, ~apt~, etc., ~nix~ can /seem/ daunting/impenetrable. These section should help demystify how to use ~nix~ as a drop in replacement for how most folks approach using the aforementioned tools.
+
+- [[#-searching-for-packages][Searching for Packages]]
+- [[#-imperitive-package-management][Imperitive Package Management]]
+  - how most users manage packages, typically by invoking an install command with the package manager of choice.
+- [[#-declaritive-package-management][Declaritive Package Management]]
+  - A more portable, version controll-able approach that declares the desired end state. Great for using the same configuration across multiple machines.
+- [[#-temporary-packages][Temporary Packages]]
+  - Excellent for testing new versions of packages, or entirely new tools, without comitting to system level changes that could impact other parts of the system.
+
+** 🔍 Searching for Pacakges
+
+#+begin_quote
+/Question: I want to install pacakges, but I'm not sure what is avalible/
+#+end_quote
+
+Search for packages via:
+- The web UI @ https://search.nixos.org/
+- Using the cli:
+  #+begin_src bash
+  nix-env -qaP 'some_package' 
+  #+end_src
+
+  - ~-q~: Stands for "query". Used to query the Nix store for /installed/ pacakges
+  - ~-a~: Stands for "avalible". Used to search for /avalible packages/ to be installed
+  - ~-P~: Includes the attribute path of the package. This result is often used with the ~-A~ flag during installs to more precisely target the desired package.
```

Comment:
I can see how this makes even me a little wary...not that this is "hard" per say, but remembering this when we previously had `brew search` makes it harder for me to want to jump to this tool

Therefore, I have a suggestion. Add this as a subsection:

```org
*** Simplify it
Add this alias so you don't need to remember the flags:

#+begin_src bash
alias nix-search="nix-env -qaP"
#+end_src
```

Except, d

Reply:
I thought about this at time of writing. The mention means I should include it.

Reply:
To be perfectly honest though, it makes me want to write a wrapper utility to nix so I can use it more like brew

The aliases are close enough